tempesta_fw/cache.c

```diff
@@ -190,6 +191,58 @@ tfw_cache_msg_cacheable(TfwHttpReq *req)
 	return __cache_method_test(req->method);
 }
 
+/*
+ * Match the given request against strings specified in "location",
+ * "cache_bypass", and "cache_fulfil" directives and according to
+ * match operators specified in the directives.
+ *
+ * Return the constant that identifies the action on a request.
+ * Currently the actions are "bypass", "fulfill", or "default".
+ */
+tfw_stmt_t
+tfw_cache_action(TfwHttpReq *req)
+{
+	TfwCfgLocation *loc;
+	TfwCfgCacheMatch *cam;
+
+	loc = tfw_location_match(&req->uri_path);
+	cam = tfw_camatch_match(loc, &req->uri_path);
+	if (cam)
+		return cam->stmt;
+	return TFW_D_CACHE_DEFAULT;
+}
```

Comment:
Well, I think the things should be done in different way. Now we're doing foundation for #471. If you see at the configuration example, then you notice that there is some _context_, which defines how to process HTTP messages (cache or not, service from local database and so on - just keep in mind `location` directive of Nginx). This means that at least `TfwHttpReq` (but probably `TfwHttpResp` as well) should have a pointer to the context.

For now _context_ means `TfwLocation`, but #471 will introduce _vhost_, the real _context_. Probably you'll find useful to define `TfwVhost` now and the structure will contain array of `TfwLocation`s.

Thus it seems that http.c should call vhost.c (i.e. the function should be moved either to http.c or vhost.c) to determine appropriate vhost for a request. Next, cache.c will also call vhost.[ch] (or just read `TfwVhost` members directly) to get caching policy.